services/apps/src/apps/service/apps/de/admin.clj

```diff
@@ -91,8 +91,10 @@
 (defn update-app
   "This service updates high-level details and labels in an App, and can mark or unmark the app as
    deleted or disabled in the database."
-  [{app-id :id :as app}]
+  [{app-name :name app-id :id :as app}]
   (validate-app-existence app-id)
+  (when-not (nil? app-name)
+    (av/validate-app-name app-name app-id (workspace-beta-app-category-id)))
```

Comment:
Should this be inside the transaction? I'm not sure how the transaction macro deals with non-SQL errors, so I don't know how the error-throwing would interact there, but if it can be that could help the potential (very rare) race condition of one being inserted between this and the transaction updating the app.